ostd/src/task/mod.rs

```diff
@@ -159,6 +159,7 @@ impl TaskOptions {
                 .take()
                 .expect("task function is `None` when trying to run");
             task_func();
+            drop(current_task);
```

Comment:
Add a comment to highlight the significance of this drop.

```rust

            // Manually drop all the on-stack variables to prevent memory leakage!
            // This is needed because `scheduler::exit_current()` will never return.
            drop(current_task);

            scheduler::exit_current();
```

I think the return value of the `exit_current` method should be changed to `!`.